x11rb-protocol/src/utils.rs

```diff
@@ -11,70 +11,13 @@
 
 #[cfg(all(feature = "std", unix))]
 mod raw_fd_container {
-    use std::mem::forget;
-    use std::os::unix::io::{AsRawFd, IntoRawFd, RawFd};
+    use std::os::unix::io::OwnedFd;
 
     /// A simple wrapper around RawFd that closes the fd on drop.
     ///
     /// On non-unix systems, this type is empty and does not provide
     /// any method.
-    #[derive(Debug, Hash, PartialEq, Eq)]
-    pub struct RawFdContainer(RawFd);
-
-    impl Drop for RawFdContainer {
-        fn drop(&mut self) {
-            let _ = nix::unistd::close(self.0);
-        }
-    }
-
-    impl RawFdContainer {
-        /// Create a new `RawFdContainer` for the given `RawFd`.
-        ///
-        /// The `RawFdContainer` takes ownership of the `RawFd` and closes it on drop.
-        pub fn new(fd: RawFd) -> Self {
-            RawFdContainer(fd)
-        }
-
-        /// Tries to clone the `RawFdContainer` creating a new FD
-        /// with `dup`. The new `RawFdContainer` will take ownership
-        /// of the `dup`ed version, whereas the original `RawFdContainer`
-        /// will keep the ownership of its FD.
-        pub fn try_clone(&self) -> Result<Self, std::io::Error> {
-            Ok(Self::new(nix::unistd::dup(self.0)?))
-        }
-
-        /// Get the `RawFd` out of this `RawFdContainer`.
-        ///
-        /// This function would be an implementation of `IntoRawFd` if that were possible. However, it
-        /// causes a conflict with an `impl` from libcore...
-        pub fn into_raw_fd(self) -> RawFd {
-            let fd = self.0;
-            forget(self);
-            fd
-        }
-
-        /// Consumes the `RawFdContainer` and closes the wrapped FD with
-        /// the `close` system call.
-        ///
-        /// This is similar to dropping the `RawFdContainer`, but it allows
-        /// the caller to handle errors.
-        pub fn close(self) -> Result<(), std::io::Error> {
-            let fd = self.into_raw_fd();
-            nix::unistd::close(fd).map_err(|e| e.into())
-        }
-    }
-
-    impl<T: IntoRawFd> From<T> for RawFdContainer {
-        fn from(fd: T) -> Self {
-            Self::new(fd.into_raw_fd())
-        }
-    }
-
-    impl AsRawFd for RawFdContainer {
-        fn as_raw_fd(&self) -> RawFd {
-            self.0
-        }
-    }
+    pub type RawFdContainer = OwnedFd;
 }
 
 #[cfg(not(all(feature = "std", unix)))]
```

Comment:
Line 31 in the new code: The non-unix `RawFdContainer` likely shouldn't derive Hash, PartialEq, and Eq since this is also no longer the case on unix. (At least I guess so from the change in x11rb-protocol)

Reply:
Fixed!